app/components/dashboard/collection_component.rb

```diff
@@ -23,5 +25,10 @@ def deposits
       policy = WorkPolicy.new(user: current_user, user_with_groups: user_with_groups)
       policy.authorized_scope(collection.works.limit(5), as: :depositor)
     end
+
+    sig { returns(T::Boolean) }
+    def deposit_blocked?
+      NO_DEPOSIT_STATUS.include? collection.state
```

Comment:
Two questions:

1. Are you open to switching the logic here such that we could replace the `unless` in the component ERB with an `if`?
1. Would you consider moving this determination into the `Collection` model? So the code here, together with :point_up:, could look something like:

```ruby
def deposit_allowed?
  collection.allows_deposit?
end
```

That way the collection can own the logic governing whether deposits are allowed or not and the component doesn't need knowledge of the collection's internal state.

Reply:
Seems like this should be in the collection policy maybe?

Reply:
Good suggestion, on it.

Reply:
I dig that.

Reply:
On reflection, it looks like the work policy would be the place.  I'd do `allows? :create?, Work.new(collection: collection)`  This is what we already do in WorkController#create